rules/integrations/gcp/credential_access_gcp_kubernetes_secrets_enumeration.toml

```diff
@@ -0,0 +1,41 @@
+[metadata]
+creation_date = "2021/06/06"
+maturity = "production"
+updated_date = "2021/06/06"
+
+[rule]
+author = ["Elastic","Austin Songer"]
+description = """
+Identifies the unauthorized enumeration of kubernetes secrets by a user.
+"""
+from = "now-20m"
+index = ["filebeat-*", "logs-gcp*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "GCP Kubernetes Secrets Enumeration"
+note = """## Config
+
+The GCP Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://cloud.google.com/kubernetes-engine/docs/how-to/audit-logging",
+]
+risk_score = 47
+rule_id = "90b6cf41-801b-4979-9233-e9a234257cd9"
+severity = "medium"
+tags = ["Elastic", "Cloud", "GCP", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:(googlecloud.audit or gcp.audit) and event.action:io.k8s.core.v*.secrets.get and event.outcome:success
```

Comment:
I could not find any reference for this event action on google, can you provide us a sample event?

Reply:
```
"query": {
    "bool": {
      "must": [],
      "filter": [
        {
          "bool": {
            "must_not": {
              "bool": {
                "should": [
                  {
                    "query_string": {
                      "fields": [
                        "gcp.audit.authentication_info.principal_email"
                      ],
                      "query": "system*"
                    }
                  }
                ],
                "minimum_should_match": 1
              }
            }
          }
        },
        {
          "range": {
            "@timestamp": {
              "gte": "2021-05-04T13:29:16.495Z",
              "lte": "2021-05-05T13:29:16.495Z",
              "format": "strict_date_optional_time"
            }
          }
        },
        {
          "match_phrase": {
            "event.module": "gcp"
          }
        },
        {
          "match_phrase": {
            "gcp.audit.type": "type.googleapis.com/google.cloud.audit.AuditLog"
          }
        },
        {
          "match_phrase": {
            "gcp.audit.method_name": "io.k8s.core.v1.secrets.get"
          }
        }
      ],
      "should": [],
      "must_not": [
        {
          "match_phrase": {
            "gcp.audit.status.message": "PERMISSION_DENIED"
          }
        }
      ]
    }
  }

```